src/utils/utils_concurrency.h

```diff
@@ -92,57 +88,144 @@ static __inline unsigned char utils_mssb_index(long long value) {
 }
 
 // There is no good way to do atomic_load on windows...
-#define utils_atomic_load_acquire(object, dest)                                \
-    do {                                                                       \
-        *(LONG64 *)dest =                                                      \
-            InterlockedOr64Acquire((LONG64 volatile *)object, 0);              \
-    } while (0)
+static __inline void utils_atomic_load_acquire_u64(uint64_t *ptr,
+                                                   uint64_t *out) {
+    // NOTE: Windows cl complains about direct accessing 'ptr' which is next
+    // accessed using Interlocked* functions (warning 28112 - disabled)
+    ASSERT_IS_ALIGNED((uintptr_t)ptr, 8);
+    LONG64 ret = InterlockedCompareExchange64((LONG64 volatile *)ptr, 0, 0);
+    utils_annotate_acquire(ptr);
```

Comment:
why we need acquire after compareandswap, with full memory fence

Reply:
removed